### PR TITLE
VideoPress: Allow feature for free sites and add paid product purchase flow

### PR DESCRIPTION
**Pull request by roundhill**
VideoPress is being added as a standalone product to Jetpack! This PR changes the way VideoPress is configured to allow access for free sites (limited to 1 video, enforced on wp.com side), and adds support for purchasing the new VideoPress product from areas of the Jetpack dashboard.

There's a Calypso PR to add the product to the pricing pages and other areas of Jeptack Cloud: https://github.com/Automattic/wp-calypso/pull/56103. _It also adds a nice `Thank You` dialog that is displayed after the purchase. Without that merged, you will just see a Jetpack Free thank you message while testing this PR._

**Screenshots**
_At-a-glance dashoard now has VideoPress card:_
Screen Shot 2021-09-08 at 4 02 58 PM

_Settings->Performance VideoPress card always displays w/ upsell banner:_
Screen Shot 2021-09-08 at 4 04 23 PM

_Storage Remaining Indicator after upgrading:_
Screen Shot 2021-09-15 at 1 07 41 PM

#### Changes proposed in this Pull Request:
* Enables the VideoPress module for free Jetpack sites.
* Updates `Video hosting` terminology to be `Jetpack VideoPress` or just `VideoPress`.
* Adds upsell banners if a user is on a free site without a plan or the VideoPress upgrade.

#### Jetpack product discussion
See pxWta-Zz-p2

#### Does this pull request change what data or activity we track or use?
I believe clicks on the upgrade buttons are tracked? This isn't new but the `jetpack_videopress` product hasn't been tracked yet.

#### Testing instructions:
* Apply D66710-code to your wp.com sandbox if it hasn't been merged yet, sandbox `public-api.wordpress.com`, and enable `JETPACK__SANDBOX_DOMAIN` on your Jetpack dev site so that it points to your wp.com sandbox. You also may need to enable HTTP access on your WP.com sandbox.
* On your WordPress.com sandbox, enable the feature flag for Jetpack VideoPress in `mu-plugins/videopress/constants.php`: `define( 'VIDEOPRESS_JETPACK_VIDEO_ENABLED', true );`
* Remove all plans/products from your site so that it is considered Jetpack Free.
* Visit `Settings->Performance` or `At-a-glance` dashboard. You should be able to enable the VideoPress module. Once enabled, it should show an upsell for VideoPress product.
* Head to the calypso media library for the site and upload a video. It should be added to VideoPress.
* Now add another video, you should get a message that you need to update your plan.
* Head back to the dashboard and purchase Jetpack VideoPress. 
* You should now see a storage space indicator, and able to upload more videos. The storage indicator should increase as you add files.
* Existing paid plans that support VideoPress (Like Security) should still support VideoPress uploads.

### CODE REVIEW

**Review by jgcaruso — COMMENTED** [review]
Code looks good, as far as my knowledge for the organisation of new components.

The PR tested well when enabling the required diffs on my sandbox. Just noticed a few items that might be worth updating?

~~1. When purchasing the upgrade I was redirected to the Jetpack welcome page in Calypso. I wonder if we should send the customer to the Media Library or some other video related screen so they can see the result of what they just purchased?~~ Didn't realise the Calypso PR takes care of this

2. When in the free tier BEFORE enabling VideoPress, the text on the "At a Glance" screen doesn't show the upsell banner, although it does say "Try it for free". Just want to make sure this intentionally not showing the upsell banner.

3. When in the free tier AFTER enabling VideoPress, the VideoPress cards in "At a Glance" and "Performance" just state that I have 1 free video, but it doesn't tell me if I've used up my 1 free video. Since we have the storage used available to us (its used in the progress bar), I wonder if we could show a `0/1 free videos used` or `1/1 free videos used` label? 

4. Should we have some kind of server toggle to enable this new behaviour incase we want to wait to enable it AFTER the Jetpack version its included in is released.

**Comment by roundhill** [comment]
Added needs review label so the crew can start looking at this earlier, because:

 * Our team is new to working with Jetpack at this level, so we very well could have missed something.
 * We have a pretty tight deadline if we want to get this merged before the next release!

**Comment by roundhill** [comment]
> I like this idea, I'll work on adding that 👍

Added in 7165d45, looks like this now:
Screen Shot 2021-09-20 at 4 39 37 PM

File: projects/plugins/jetpack/_inc/client/performance/media.jsx
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/performance/media.jsx:116]: Do we need an s here? The wording feels a bit odd:

image

Maybe we don't need the `1` number?

**Review by jeherve — COMMENTED** [review]
I gave this a first try, here are my remarks below.

*************

I see the banner as soon as I connect my site to WordPress.com, even though I haven't uploaded anything:

image

*************

The banner is displayed even when in offline mode:

Screenshot 2021-09-21 at 12 09 31

I don't think it should be displayed at all in those cases.

************

It doesn't really work well when the site is connected, but no user is connected yet. The 2 banners kinda conflict with each other:

image

(To connect the site, but not the user, do not complete the connection process, hit the back button instead of approving when you get redirected to WordPress.com during the connection process)

************

> Head to the calypso media library for the site and upload a video. It should be added to VideoPress.

I can't quite get this to work. The `/media/new` API call returns a success, but without any response:

image

I get the same error even after purchasing a Jetpack VideoPress plan though.

************

I guess that's already on your list, but just in case: it looks like we'll need to update the post-checkout process in Calypso, too. Here is what I get after purchasing a Jetpack VideoPress plan:

image

image

************

File: projects/plugins/jetpack/_inc/client/performance/media.jsx
Comment (robertbpugh) [reply projects/plugins/jetpack/_inc/client/performance/media.jsx:116]: I'm good with that

**Comment by jgcaruso** [comment]
Might just be a quirk of how I disconnected, but when I was testing the "Jetpack is enabled, but not authorized" state, I saw the storage progress bar instead of the disabled state
Screen Shot 2021-09-22 at 3 51 52 PM

But otherwise, the `connect` buttons now appear when jetpack is enabled but not authorized.

**Comment by roundhill** [comment]
> Might just be a quirk of how I disconnected, but when I was testing the "Jetpack is enabled, but not authorized" state, I saw the storage progress bar instead of the disabled state

Ok, I don't think the progress bar cares if you are connected are not. Should I hide it if you're not connected?

**Comment by jeherve** [comment]
> @jeherve did you apply the diffs in the testing instructions? Looks to me that you aren't getting the API response from the sandbox:

I did, yes, but I'm happy to try again if needed.

**Comment by roundhill** [comment]
> I did, yes, but I'm happy to try again if needed.

Sure, and maybe ensure that `public-api.wordpress.com` is sandboxed in your hosts file so that the js api calls hit it.

**Review by jeherve — COMMENTED** [review]
I'm afraid I'm still running into the same issues. This may be something with my setup so I'll let others chime in and review.

I did note, although it may not be a blocker for this PR (this can be addressed in a follow-up PR) that the VideoPress feature wasn't made available in the old module list (click on "Modules" in the footer of the Jetpack dashboard, or head to `wp-admin/admin.php?page=jetpack_modules`)
Screenshot 2021-09-27 at 18 04 46

This may be linked to the header in the VideoPress module:
https://github.com/Automattic/jetpack/blob/74d0505819b31d77294c36bc5b10acb60410c891/projects/plugins/jetpack/modules/videopress.php#L6-L12
As well as because of the special status of the module in the list:
https://github.com/Automattic/jetpack/blob/5c73eef31002b46d5a0ce3e23578e510a40e59fc/projects/plugins/jetpack/class.jetpack-modules-list-table.php#L102-L116

**Comment by roundhill** [comment]
> the VideoPress feature wasn't made available in the old module list

Thanks @jeherve, didn't know that existed ;) Will create a separate PR

File: projects/plugins/jetpack/_inc/client/lib/plans/constants.js
Comment (samiff) [inline projects/plugins/jetpack/_inc/client/lib/plans/constants.js:339]: These appear nested under the "DEPRECATED" comment on L323, is that correct?

**Review by samiff — CHANGES_REQUESTED** [review]
Testing steps worked well for me 👍 

> Head back to the dashboard and purchase Jetpack VideoPress.

When checking out I only had a "billed monthly" option, is there not a yearly plan option?

> `PLAN_JETPACK_VIDEOPRESS`

Also, is there a reason we didn't follow lead of backup/security plans for this as well? Such as:

```
PLAN_JETPACK_SECURITY_T1_YEARLY
PLAN_JETPACK_SECURITY_T1_MONTHLY
PLAN_JETPACK_SECURITY_T2_YEARLY
PLAN_JETPACK_SECURITY_T2_MONTHLY
```

File: projects/plugins/jetpack/_inc/client/lib/plans/constants.js
Comment (roundhill) [reply projects/plugins/jetpack/_inc/client/lib/plans/constants.js]: Not correct! Fixed in e23cab2.

**Comment by roundhill** [comment]
> When checking out I only had a "billed monthly" option, is there not a yearly plan option?

I think these `Upgrade` buttons on the dashboard take you directly to checkout, with only one type of billing option. At least it does that for me with the other `Upgrade` buttons like on the `Backup` card. It does support annual billing.

> Also, is there a reason we didn't follow lead of backup/security plans for this as well? Such as:

I was following the same naming conventions as the other products, like Anti-Spam:

`PLAN_JETPACK_ANTI_SPAM`
`PLAN_JETPACK_ANTI_SPAM_MONTHLY`

**Review by samiff — APPROVED** [review]

**Comment by samiff** [comment]
> I was following the same naming conventions as the other products, like Anti-Spam:

Got it, I was curious because I saw mentioned in the linked P2:

> If possible, a user should be able to upgrade from Paid Tier 1 to Paid Tier 2.

But didn't drill down to see where we landed on that exactly in related discussions.

If the plans aren't any cause for concern currently, should be able to merge this so we can get it in the beta. Great work 🎉

**Comment by roundhill** [comment]
> But didn't drill down to see where we landed on that exactly in related discussions

Ah yes, I'll update those requirements. We ended up simplifying and just doing one upgrade tier for the product (1TB of storage)

**Comment by robertbpugh** [comment]
> > When checking out I only had a "billed monthly" option, is there not a yearly plan option?
> 
> I think these `Upgrade` buttons on the dashboard take you directly to checkout, with only one type of billing option. At least it does that for me with the other `Upgrade` buttons like on the `Backup` card. It does support annual billing. I could be missing something though?

Can you make the one option go to the yearly plan? That would be consistent with other products. We're working on making the monthly/yearly selection easier.

**Comment by roundhill** [comment]
> Can you make the one option go to the yearly plan? That would be consistent with other products. We're working on making the monthly/yearly selection easier.

Done, just needed to update the redirect url. (Outside of this PR)

**Comment by roundhill** [comment]
> That would be consistent with other products.

@robertbpugh just a heads up that the `Backup` Upgrade button uses monthly billing.

**Comment by robertbpugh** [comment]
Okay we'll circle back to make the Backup button yearly.

**Pull request merged**
